packages/components/src/useStep.js

```diff
@@ -0,0 +1,11 @@
+import { useContext, createContext } from 'react'
+
+const StepContext = createContext(0)
+
+export const useStep = () => useContext(StepContext)
+
+export const Step = props => (
+  <StepContext.Provider value={props.index}>
+    {props.children}
+  </StepContext.Provider>
```

Comment:
IMO: If we can find a solution that doesn't require this `<Step>` component, we should go with it.